spec/lib/google_maps_geocoder_spec.rb

```diff
@@ -4,74 +4,74 @@
 
 require "#{File.dirname(__FILE__)}/../spec_helper"
 # rubocop:disable Metrics/BlockLength
-describe GoogleMapsGeocoder do
-  before(:all) do
-    # rubocop:disable Lint/RedundantCopDisableDirective, Style/RedundantBegin
-    begin
-      @exact_match = GoogleMapsGeocoder.new('White House')
-    rescue SocketError
-      @no_network  = true
-    rescue RuntimeError
-      @query_limit = true
-    end
-    # rubocop:enable Lint/RedundantCopDisableDirective, Style/RedundantBegin
-  end
-
-  before(:each) do
-    pending 'waiting for a network connection' if @no_network
-    pending 'waiting for query limit to pass' if @query_limit
-  end
-
+RSpec.describe GoogleMapsGeocoder, silence_logger: true do
   describe '#new' do
     context 'with "White House"' do
-      subject { @exact_match }
+      subject(:geocoder) do
+        GoogleMapsGeocoder.new('White House')
+      rescue SocketError
+        pending 'waiting for a network connection'
+      rescue GoogleMapsGeocoder::GeocodingError
+        pending 'waiting for query limit to pass'
+      end
 
-      it { should be_exact_match }
+      it('should be an exact match', silence_logger: false) {
+        a_quiet_logger = Logger.new(IO::NULL)
```

Comment:
Clever approach, but it doesn't seem to work. I removed the mocking from `spec_helper` in a test branch here...

https://github.com/FoveaCentral/google_maps_geocoder/pull/93/files

But notice the log output (under "Run tests") still goes to standard out...

https://github.com/FoveaCentral/google_maps_geocoder/runs/4313718453?check_suite_focus=true

Presumably because the mocking happens too late in the lifecycle—ie: the subject, and with it the logger, is already instantiated by the time the `it` block runs.

Unless something else occurs to you, we'll have to go with our previous approach. In the meantime, I'll clean-up the code with that assumption in mind.

Reply:
Clean-up in this branch, I mean.